File: week8/pureAbstract.java
interface Shape
{
    float area();
    void dimensions();
}
class Rectangle implements Shape
{
    float a,b;
    Rectangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return a*b;
    }
    public void dimensions()
    {
        System.out.println("Dimensions : "+a+" X "+b);
    }
}
class Triangle implements Shape
{
    float a,b;
    Triangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return (float)0.50*a*b;
    }
    public void dimensions()
    {
        System.out.println("Dimensions : "+a+" X "+b);
    }
}
public class Main
{
    public static void main(String[] args)
    {
        Rectangle obj=new Rectangle(16,25);
        Triangle obj1=new Triangle(10,6);
        System.out.println("The area of a Rectangle : "+obj.area());obj.dimensions();
        System.out.println("The area of a Triangle : "+obj1.area());obj1.dimensions();
    }
}

File: week8/pureAbstract.cpp
#include<iostream>

using namespace std;

class Shapes
{
    private:
    virtual float area()=0;
};
class Triangle: public Shapes
{
    public:
    float b,h;
    Triangle(float base,float height)
    {
        b=base;
        h=height;
    }
    float area()
    {
        return 0.5*b*h;
    }
};
class Circle: public Shapes
{
    public:
    float r;
    Circle(float radius)
    {
        r=radius;
    }
    float area()
    {
        return 3.14*r*r;
    }
};
int main()
{
    Triangle tri(6,4);
    Circle cir(8.5);
    cout<<"The area of Triangle : "<<tri.area()<<endl;
    cout<<"The area of a Circle: "<<cir.area()<<endl;
    return 0;
}
